data/data.go

```diff
@@ -31,6 +31,91 @@ type EffxYaml struct {
 	FilePath string
 }
 
+// given a absolute path (example: /runner/folder/root/data/effx.yaml)
+// and given a workingDir (example: root)
+// it will return data/effx.yaml
+func parseRelativePathFromAbsolutePath(absoluteDir, workingDir string) string {
+	// if working directory does not end with a slash, add it
+	if strings.LastIndex(workingDir, "/") != len(workingDir)-1 {
+		workingDir += "/"
+	}
+
+	res := strings.Split(absoluteDir, workingDir)
+	if len(res) > 1 {
+		return res[1]
+	}
+	return absoluteDir
+}
+
+// given an absolute path, this function will find the root directory
+// (directory with a .git file)
+func findGitRootDirectory(absoluteDir string) (string, error) {
+	pathDir := filepath.Dir(absoluteDir)
+
+	for pathDir != "" {
+		pathDir = filepath.Clean(pathDir)
+		files, err := ioutil.ReadDir(pathDir)
+
+		if err != nil {
+			return "", err
+		}
+
+		for _, file := range files {
+			if !file.IsDir() {
+				if strings.Contains(file.Name(), ".git") {
+					return filepath.Base(pathDir), nil
+				}
+			}
+		}
+
+		pathDir = filepath.Join(pathDir, "..")
+	}
+
+	return absoluteDir, nil
+}
+
+func getRepoName() string {
+	repoNameList := []string{
+		"CIRCLE_PROJECT_REPONAME",
+		"GITHUB_REPOSITORY",
+		"CI_PROJECT_NAME",
+		"SEMAPHORE_GIT_URL",
```

Comment:
From: https://docs.semaphoreci.com/ci-cd-environment/environment-variables/#semaphore_git_url

An example value for `SEMAPHORE_GIT_URL` is `http://git@github.com:semaphoreci/toolbox.git`. I think it may be better to parse `SEMAPHORE_GIT_REPO_SLUG` for the repo name.

Reply:
why not SEMAPHORE_PROJECT_NAME? slugs probably have ugly dashes right?